packages/app/src/vis-packs/nexus/hooks.ts

```diff
@@ -1,10 +1,38 @@
-import type { ScaleType } from '@h5web/shared';
+import type { GroupWithChildren, ScaleType } from '@h5web/shared';
 import { isDefined } from '@h5web/shared';
+import { useContext } from 'react';
 
+import { ProviderContext } from '../../providers/context';
 import { useDatasetValues } from '../core/hooks';
 import type { AxisMapping } from '../core/models';
-import type { AxisDatasetMapping } from './models';
-import { getDatasetLabel } from './utils';
+import type { AxisDatasetMapping, NxData } from './models';
+import {
+  assertNxDataGroup,
+  findAssociatedDatasets,
+  findErrorsDataset,
+  findSignalDataset,
+  findTitleDataset,
+  getDatasetLabel,
+  getSilxStyle,
+} from './utils';
+
+export function useNxData(group: GroupWithChildren): NxData {
+  const { attrValuesStore } = useContext(ProviderContext);
+
+  assertNxDataGroup(group, attrValuesStore);
```

Comment:
`useNxData` is now a hook so I can get `attrValuesStore` and pass it to `assertNxDataGroup`, which passes it to `isNxDataGroup`.